spec-tests/src/test/java/org/candlepin/spec/pools/PoolSpecTest.java

```diff
@@ -462,69 +460,64 @@ public void shouldIncludeCalculatedAttributesWhenFetchingPools() {
             .containsEntry("compliance_type", "Standard");
     }
 
-    // @Test
-    // public void shouldAllowFetchingCDNByPoolId() {
-    //     // Impl note:
-    //     // This test is currently blocked on a limitation with CDNs in that we can perform CRUD
-    //     // operations on them (as they are first-class objects for some reason), but the only way
-    //     // to assign a CDN to a pool is via manifest import or pool refresh -- neither of which are
-    //     // implementated at the time of writing.
-
-    //     ApiClient adminClient = ApiClients.admin();
-    //     OwnerDTO owner = this.createOwner(adminClient);
+    @Test
+    @OnlyInHosted
+    public void shouldAllowFetchingCDNByPoolId() {
+        ApiClient adminClient = ApiClients.admin();
+        OwnerDTO owner = this.createOwner(adminClient);
+        HostedTestApi upstreamClient = adminClient.hosted();
 
-    //     CdnDTO cdn = new CdnDTO()
-    //         .name(StringUtil.randomSuffix("test_cdn"))
-    //         .label("Test CDN")
-    //         .url("https://cdn.test.com");
+        CdnDTO cdn = new CdnDTO()
+            .name(StringUtil.random("test_cdn"))
+            .label(StringUtil.random("Test CDN"))
+            .url("https://cdn.test.com");
```

Comment:
In PR-3509 I added a Cdn builder and you can copy it over and make this a one liner with `CdnDTO cdn = Cdns.random()`.


https://github.com/candlepin/candlepin/pull/3509/files#diff-2028afa08357f43806ba504c152be9d7e173882e769f29f9b1ab9e65557c7970

Reply:
done